pandas/tests/indexing/test_loc.py

```diff
@@ -2702,6 +2703,27 @@ def test_loc_indexer_length_one(self):
         df.loc[np.array([True], dtype=np.bool_), ["a"]] = df["b"].copy()
         tm.assert_frame_equal(df, expected)
 
+    def test_loc_setitem_bool_mask_tzaware_scalar_with_expansion(self):
+        # GH#55423
+
+        df = DataFrame([{"id": 1}, {"id": 2}, {"id": 3}])
+        _time = datetime.fromtimestamp(1695887042, tz=timezone.utc)
+
+        df.loc[df.id >= 2, "time"] = _time
+
+        assert isinstance(df["time"].dtype, pd.DatetimeTZDtype)
+        assert str(df["time"].dtype.tz) == "UTC"
+
+        expected_time = Timestamp(_time)
+        expected = DataFrame(
+            {
+                "id": [1, 2, 3],
+                "time": [pd.NaT, expected_time, expected_time],
```

Comment:
```suggestion
                "time": [pd.NaT, _time, _time],
```

The issue is about the Python datetime type - we don't want to wrap it with `Timestamp`